apluslms_roman/validator.py

```diff
@@ -0,0 +1,85 @@
+import yaml
+from jsonschema import validate
+import re
+import os
+#from operator import itemgetter
+import logging
+
+logger = logging.getLogger(__name__)
+
+#mystinen lista schema directoryja - course.yamlissa jossain kentässä, joka sijaitsee käyttäjän paketin juuressa
+def get_folders(config):
+    folders = config.__config__.get("custom_schema_locations", [])
+    dir_, _ = os.path.split(__file__)
+    folders.append(os.path.join(dir_, "schemas"))
+    return folders
+
+
+#idea = { schema_name: {(major, minor): location}}
+schema_index = {}
+
+class Validator:
+    def __init__(self, config):
+        logger.info(" Creating validator.")
+        self.folders = get_folders(config)
+        logger.info(" Validator ready.")
+
+    def validate(self, filename, schema, major=1, minor=float('inf')):
+        logger.info(" Preparing to validate %s...", filename)
+        logger.info(" Locating all valid schemas...")
+        self.find_all_matches(schema)
+        logger.info(" Locating matching version v%d...", major)
+        serial, ext = self.find_newest(schema, major, minor)
```

Comment:
this is required only when minor is not known `version: '3'`. If minor is known, then test `(major, minor) in schema_index[schema]` is sufficient.